sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -557,6 +557,12 @@ CURLcode CurlConnection::SendBuffer(
     size_t bufferSize,
     Context const& context)
 {
+  // Once you've shutdown the connection, we can't send any more data (although we can continue to
+  // receive).
+  if (IsShutdown())
```

Comment:
Sockets allow you to shutdown the socket for sends while keeping the socket open for receives.  Is there a reason we don't want that behavior here?

https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-shutdown

After your change, Where do we shutdown the socket?

Reply:
We shut down the socket when the connection is released (via curl_easy_shutdown).

With my change, if a Connection: close is set, we won't put the connection back into the connection pool, it will just be normally freed.

That allows all the normal operations to be performed on the socket, most importantly, if the operation is a "put" operation and the server responds with "Connection: close", if we called shutdown to block sends, then the "put" would fail.

The new behavior is consistent with the RFC mentioned in the original bug report: Connection: close means "don't persist the connection after this HTTP transaction".

Reply:
> I don't get it. How does the change in this PR prevent the connection being moved back into the pool?

When a connection is marked as "Shutdown", it won't be put back into the pool (see `CurlConnectionPool::MoveConnectionBackToPool` which ignores connections marked as `Shutdown`).

What this PR does is to prevent the underlying socket used in the connection from being shut down, which has the effect of allowing the connection to be processed normally (sends and receives continue to work).

Based on the text you quoted from the RFC, all `Connection: close` is supposed to do is to prevent a connection from being re-used, which is what the `CurlConnection::Shutdown` function does.

Reply:
@RickWinter, I am going to resolve this comment thread now so that I can merge this PR.
If you'd like more follow up your question, please create a work item so that we can follow up on it later.